opm/simulators/wells/WellState.cpp

```diff
@@ -102,12 +99,15 @@ void WellState::initSingleWell(const std::vector<double>& cellPressures,
     const int num_perf_this_well = well_info->communication().sum(well_perf_data_[w].size());
     this->perfpress_.add(well.name(), std::vector<double>(num_perf_this_well, -1e100));
     this->perfrates_.add(well.name(), std::vector<double>(num_perf_this_well, 0));
-    if ( num_perf_this_well == 0 ) {
-        // No perforations of the well. Initialize to zero.
-        bhp_[w] = 0.;
-        thp_[w] = 0.;
+    this->bhp_.add(well.name(), 0.0);
+    this->thp_.add(well.name(), 0.0);
+    if ( well.isInjector() )
+        this->temperature_.add(well.name(), well.injectionControls(summary_state).temperature);
+    else
+        this->temperature_.add(well.name(), 273.15 + 15.56); // standard condition temperature
+
```

Comment:
Why wasn't this needed before?

Reply:
The well temperature has always been initialized to 273.15 + 15.56  - no change here.